sentry-android-core/src/main/java/io/sentry/android/core/ViewHierarchyEventProcessor.java

```diff
@@ -0,0 +1,173 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+import androidx.annotation.NonNull;
+import io.sentry.Attachment;
+import io.sentry.EventProcessor;
+import io.sentry.Hint;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.android.core.internal.gestures.ViewUtils;
+import io.sentry.protocol.ViewHierarchy;
+import io.sentry.protocol.ViewHierarchyNode;
+import io.sentry.util.Objects;
+import java.io.BufferedWriter;
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** ViewHierarchyEventProcessor responsible for taking a snapshot of the current view hierarchy. */
+@ApiStatus.Internal
+public final class ViewHierarchyEventProcessor implements EventProcessor {
+
+  @SuppressWarnings("CharsetObjectCanBeUsed")
+  private static final @NotNull Charset UTF_8 = Charset.forName("UTF-8");
+
+  private final @NotNull SentryAndroidOptions options;
+
+  public ViewHierarchyEventProcessor(final @NotNull SentryAndroidOptions options) {
+    this.options = Objects.requireNonNull(options, "SentryAndroidOptions is required");
+  }
+
+  @Override
+  public @NotNull SentryEvent process(final @NotNull SentryEvent event, final @NotNull Hint hint) {
+    if (!event.isErrored()) {
+      return event;
+    }
+
+    if (!options.isAttachViewHierarchy()) {
+      options.getLogger().log(SentryLevel.DEBUG, "attachViewHierarchy is disabled.");
+      return event;
+    }
+
+    final @Nullable Activity activity = CurrentActivityHolder.getInstance().getActivity();
+    if (activity == null) {
+      options.getLogger().log(SentryLevel.INFO, "Missing activity for view hierarchy snapshot.");
+      return event;
+    }
+
+    final @Nullable Window window = activity.getWindow();
+    if (window == null) {
+      options.getLogger().log(SentryLevel.INFO, "Missing window for view hierarchy snapshot.");
+      return event;
+    }
+
+    final @Nullable View decorView = window.peekDecorView();
+    options.getLogger().log(SentryLevel.INFO, "Missing decor view for view hierarchy snapshot.");
```

Comment:
This log should be inside of the if below. Otherwise, it will be always logged.

Reply:
nice catch, thanks!